aws/resource_aws_fsx_lustre_file_system.go

```diff
@@ -89,7 +89,6 @@ func resourceAwsFsxLustreFileSystem() *schema.Resource {
 			"storage_capacity": {
 				Type:         schema.TypeInt,
 				Required:     true,
-				ForceNew:     true,
```

Comment:
lets add a check to reduce the size, its probably not a valid case and we should trigger a recreate on that case.

Reply:
Hi @DrFaust92 

I am happy to help out here. I presume what you want is a `CustomDiff` as per `github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff` ?

Reply:
This is what I was going to add (with tests):

```
func resourceFsxLustreFileSystemSchemaCustomizeDiff(_ context.Context, d *schema.ResourceDiff, meta interface{}) error {
	// we want to force a new resource if the new storage capacity is less than the old one
	if d.HasChange("storage_capacity") {
		o, n := d.GetChange("storage_capacity")
		if n.(int) < o.(int) {
			if err := d.ForceNew("storage_capacity"); err != nil {
				return err
			}
		}
	}

	return nil
}
```